Intersect.Client/Entities/Entity.cs

```diff
@@ -1387,60 +1387,93 @@ public virtual void DrawName(Color textColor, Color borderColor = null, Color ba
         }
 
         // NpcTags
-        public virtual void DrawNpcTag()
+        public virtual void DrawTag()
         {
-            // Lets grab the size of the Npc's name, we will use both, X and Y for this.
-            var textSize = Graphics.Renderer.MeasureText(this.Name, Graphics.EntityNameFont, 1);
-            // String that will be used with the switch cases.
-            string NpcTagFile;
-            switch (Type)
-                {
-                    case -1: //When entity has a target (showing aggression).
-                        NpcTagFile = Options.Npc.AggressiveNpcTag;
+            // We make sure that the entity isn't a Player.
+            if (Options.Npc.ShowTags && this.GetType() != typeof(Player))
+            {
+                // strings used for the sprites file names.
+                string npcName = this.Name;
+                string npcTagFile;
```

Comment:
Declare strings together.

Reply:
all variables are declared at the beggining of the function now 👍🏽